src/lib.rs

```diff
@@ -31,6 +31,7 @@ pub use crate::bounds::Bounded;
 #[cfg(any(feature = "std", feature = "libm"))]
 pub use crate::float::Float;
 pub use crate::float::FloatConst;
+pub use crate::float::TotalOrder;
```

Comment:
I think the best way to signal that this trait is only intended for floats is to *not* re-export it here, so you have to `use num_traits::float::TotalOrder`.

(And if people "abuse" it beyond that intent, it's not really our problem.)

Reply:
Agree -- limiting exposure decreases probability of abuse.